pandas/_libs/ops.pyx

```diff
@@ -160,17 +160,19 @@ def vec_compare(ndarray[object] left, ndarray[object] right, object op) -> ndarr
         for i in range(n):
             x = left[i]
             y = right[i]
-
-            if checknull(x) or checknull(y):
+            if x is None and y is None:
+                result[i] = True
+            elif checknull(x) or checknull(y):
                 result[i] = True
             else:
                 result[i] = PyObject_RichCompareBool(x, y, flag)
     else:
         for i in range(n):
             x = left[i]
             y = right[i]
-
-            if checknull(x) or checknull(y):
+            if x is None and y is None:
```

Comment:
this is contrary to what we do with nulls generally, does anything break when you make this change (only)?

Reply:
AFAIK, this does not break any tests. However, I am starting to think that #35565 is not actually a bug, but intended behavior. From the [docs](https://pandas.pydata.org/pandas-docs/stable/user_guide/missing_data.html#values-considered-missing): "One has to be mindful that in Python (and NumPy), the nan's don’t compare equal, but None's do. Note that pandas/NumPy uses the fact that np.nan != np.nan, and treats None like np.nan."

I see the same behavior in `scalar_compare`:
https://github.com/pandas-dev/pandas/blob/1269bc6b7f3f3f4fc253cfa11f795d6bccda55a0/pandas/_libs/ops.pyx#L74-L86

I found this related issue: https://github.com/pandas-dev/pandas/issues/34975

Should we close this PR and the issue, then?